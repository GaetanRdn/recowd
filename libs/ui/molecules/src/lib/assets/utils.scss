@function map-get-deep($map, $level, $subLevel) {
  @return map-get($map, $level, $subLevel);
}

@mixin use-typography($typography) {
  @each $key in font-size, font-family, font-weight, line-height {
    #{$key}: map-get($typography, $key);
  }
}
