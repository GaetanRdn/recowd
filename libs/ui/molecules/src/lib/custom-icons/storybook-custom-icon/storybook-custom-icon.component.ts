import { ChangeDetectionStrategy, Component } from '@angular/core';
import { CustomIconComponent } from '@recowd/ui-atoms';

@Component({
  selector: 'rc-storybook-custom-icon',
  standalone: true,
  imports: [CustomIconComponent],
  templateUrl: './storybook-custom-icon.component.html',
  changeDetection: ChangeDetectionStrategy.OnPush,
})
export class StorybookCustomIconComponent extends CustomIconComponent {
  public override readonly name = 'storybook';
}
