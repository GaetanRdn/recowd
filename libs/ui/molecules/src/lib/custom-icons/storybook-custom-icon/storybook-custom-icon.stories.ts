import { Meta, moduleMetadata, Story } from '@storybook/angular';
import { StorybookCustomIconComponent } from './storybook-custom-icon.component';

export default {
  title: 'molecules/custom-icons/storybook-custom-icon',
  parameters: {
    jest: 'storybook-custom-icon',
  },
  decorators: [
    moduleMetadata({
      imports: [StorybookCustomIconComponent],
    }),
  ],
} as Meta<StorybookCustomIconComponent>;

const Template: Story<StorybookCustomIconComponent> = () => ({
  template: `<rc-storybook-custom-icon></rc-storybook-custom-icon>`,
});

export const Default = Template.bind({});
