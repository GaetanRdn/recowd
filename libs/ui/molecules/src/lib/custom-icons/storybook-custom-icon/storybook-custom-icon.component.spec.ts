import { TestBed, waitForAsync } from '@angular/core/testing';

import { StorybookCustomIconComponent } from './storybook-custom-icon.component';
import { Component } from '@angular/core';
import { TemplateLookup } from '@recowd/test/utils';

describe('StorybookCustomIconComponent', () => {
  let templateLookup: TemplateLookup<HostComponent>;

  beforeEach(waitForAsync(() => {
    TestBed.configureTestingModule({
      declarations: [HostComponent],
      imports: [StorybookCustomIconComponent],
    }).compileComponents();

    templateLookup = new TemplateLookup(HostComponent);
  }));

  it.each`
    width | height
    ${32} | ${undefined}
    ${16} | ${24}
  `(
    'should create with name $name, width $width and height $height',
    ({ width, height }) => {
      templateLookup.hostComponent.width = width;
      templateLookup.hostComponent.height = height;

      templateLookup.detectChanges();

      expect(templateLookup.firstChildElement).toMatchSnapshot();
    }
  );
});

@Component({
  template: ` <rc-storybook-custom-icon
    [width]="width"
    [height]="height"
  ></rc-storybook-custom-icon>`,
})
class HostComponent {
  public width!: StorybookCustomIconComponent['width'];
  public height!: StorybookCustomIconComponent['height'];
}
