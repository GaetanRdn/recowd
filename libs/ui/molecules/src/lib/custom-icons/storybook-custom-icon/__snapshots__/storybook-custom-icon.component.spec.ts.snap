// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StorybookCustomIconComponent should create with name $name, width 16 and height 24 1`] = `
<rc-storybook-custom-icon>
  <rc-custom-icon>
    <img
      alt="storybook"
      height="24"
      src="assets/images/storybook.svg"
      width="24"
    />
  </rc-custom-icon>
</rc-storybook-custom-icon>
`;

exports[`StorybookCustomIconComponent should create with name $name, width 32 and height undefined 1`] = `
<rc-storybook-custom-icon>
  <rc-custom-icon>
    <img
      alt="storybook"
      height="24"
      src="assets/images/storybook.svg"
      width="24"
    />
  </rc-custom-icon>
</rc-storybook-custom-icon>
`;
